databricks_cli/jobs/api.py

```diff
@@ -27,30 +27,33 @@ class JobsApi(object):
     def __init__(self, api_client):
         self.client = JobsService(api_client)
 
-    def create_job(self, json, headers=None):
-        return self.client.client.perform_query('POST', '/jobs/create', data=json, headers=headers)
+    def create_job(self, json, headers=None, version=None):
+        return self.client.client.perform_query('POST', '/jobs/create', data=json, headers=headers,
+                                                version=version)
 
-    def list_jobs(self, headers=None):
-        resp = self.client.list_jobs(headers=headers)
+    def list_jobs(self, expand_tasks=None, offset=None, limit=None, headers=None, version=None):
+        resp = self.client.list_jobs(expand_tasks=expand_tasks, offset=offset, limit=limit,
+                                     headers=headers, version=version)
         if 'jobs' not in resp:
             resp['jobs'] = []
         return resp
 
-    def delete_job(self, job_id, headers=None):
-        return self.client.delete_job(job_id, headers=headers)
+    def delete_job(self, job_id, headers=None, version=None):
+        return self.client.delete_job(job_id, headers=headers, version=version)
 
-    def get_job(self, job_id, headers=None):
-        return self.client.get_job(job_id, headers=headers)
+    def get_job(self, job_id, headers=None, version=None):
+        return self.client.get_job(job_id, headers=headers, version=version)
 
-    def reset_job(self, json, headers=None):
-        return self.client.client.perform_query('POST', '/jobs/reset', data=json, headers=headers)
+    def reset_job(self, json, headers=None, version=None):
+        return self.client.client.perform_query('POST', '/jobs/reset', data=json, headers=headers,
+                                                version=version)
 
     def run_now(self, job_id, jar_params, notebook_params, python_params, spark_submit_params,
-                headers=None):
+                headers=None, version=None):
         return self.client.run_now(job_id, jar_params, notebook_params, python_params,
-                                   spark_submit_params, headers=headers)
+                                   spark_submit_params, headers=headers, version=version)
 
-    def _list_jobs_by_name(self, name, headers=None):
-        jobs = self.list_jobs(headers=headers)['jobs']
+    def _list_jobs_by_name(self, name, headers=None, version=None):
```

Comment:
This will not work in API 2.1. Maybe we should disable this method in the CLI if using API 2.1, for now?

Reply:
Nevermind this. If you pass `--all` you can do it.

Reply:
Yeah, I'll print a warning if this is called without `--all`